statsd/aggregator.go

```diff
@@ -172,12 +172,31 @@ func (a *aggregator) flushMetrics() []metric {
 }
 
 func getContext(name string, tags []string) string {
-	return name + ":" + strings.Join(tags, tagSeparatorSymbol)
+	c, _ := getContextAndTags(name, tags)
+	return c
 }
 
 func getContextAndTags(name string, tags []string) (string, string) {
-	stringTags := strings.Join(tags, tagSeparatorSymbol)
-	return name + ":" + stringTags, stringTags
+	n := len(name) + len(nameSeparatorSymbol) + len(tagSeparatorSymbol)*(len(tags)-1)
```

Comment:
Doing a quick benchmark on `getContextAndTags` I get nice improvements except when we have 0 tags:

```
$> benchcmp before after
benchmark                           old ns/op     new ns/op     delta
BenchmarkGetContextAndTags0-6       64.9          118           +81.29%
BenchmarkGetContextAndTags1-6       89.0          74.0          -16.88%
BenchmarkGetContextAndTags5-6       297           157           -47.21%
BenchmarkGetContextAndTags10-6      428           259           -39.50%
BenchmarkGetContextAndTags50-6      1917          1186          -38.13%
BenchmarkGetContextAndTags100-6     3692          2358          -36.13%
```

If we catch this use case with by adding:
```
        if len(tags) == 0 {
		return name + ":", ""
	}
```

We get better performances everywhere:
```
$> benchcmp before after
benchmark                           old ns/op     new ns/op     delta
BenchmarkGetContextAndTags0-6       64.9          52.1          -19.72%
BenchmarkGetContextAndTags1-6       89.0          65.0          -26.93%
BenchmarkGetContextAndTags5-6       297           152           -48.99%
BenchmarkGetContextAndTags10-6      428           244           -43.01%
BenchmarkGetContextAndTags50-6      1917          1205          -37.14%
BenchmarkGetContextAndTags100-6     3692          2234          -39.49%
```

Could you add this edge case ?

Reply:
Nice catch! Thanks for checking the benchmarks. I have added a commit to fix the edge case.